frontend/src/lib/components/Cards/InsightCard/InsightCard.tsx

```diff
@@ -394,6 +407,36 @@ function InsightMeta({
     )
 }
 
+function TopHeading({ insight }: { insight: InsightModel }): JSX.Element {
+    const { filters, query } = insight
+
+    let insightType: InsightTypeMetadata
+
+    // check the query first because the backend still adds defaults to empty filters :/
+    if (!!query?.kind) {
+        insightType = QUERY_TYPES_METADATA[query.kind]
+    } else if (!!filters.insight) {
+        insightType = INSIGHT_TYPES_METADATA[filters.insight]
+    } else {
+        // maintain the existing default
+        insightType = INSIGHT_TYPES_METADATA[InsightType.TRENDS]
+    }
+
+    let { date_from, date_to } = filters
+    if (!!query && isInsightQueryNode(query)) {
```

Comment:
This is a little yucky...

Probably need a helper function to get the date range off any query...

Reply:
Yes, let's try to unify the schema for all queries with a start and end date as good as possible.

Reply:
Pulled the calculation out of Insight Card and into schema utils for now.